### PR TITLE
Add check for the overflow in Python bindings

### PR DESCRIPTION
**Pull request by yamaguchi1024**
Add check for ssize_t -> int32_t overflow. Add the corresponding test  to correctness/buffer.py.

### CODE REVIEW

File: python_bindings/src/PyBuffer.cpp [hunk 1]
```diff
@@ -231,7 +231,8 @@ class PyBuffer : public Buffer<> {
         std::vector<halide_dimension_t> dims;
         dims.reserve(info.ndim);
         for (int i = 0; i < info.ndim; i++) {
-            // TODO: check for ssize_t -> int32_t overflow
+            if (INT_MAX < info.shape[i] || INT_MAX < (info.strides[i] / t.bytes()))
```
Comment (steven-johnson) [inline python_bindings/src/PyBuffer.cpp]: nit: enclose all if clauses in braces, even if a single line.

**Comment by yamaguchi1024** [comment]
@steven-johnson Thanks for your comment! I updated the patch. As said in https://chezjrk.slack.com/archives/C752YMC95/p1553338460001800, I'm trying to contribute and didn't actually hit this bug. However the size of `info.shape[i]` was 64bit on my machine, so there are still a possibility of overflow here.

**Comment by steven-johnson** [comment]
I like this change, but I'd like to see a test added before we land it.

**Comment by yamaguchi1024** [comment]
Hi @steven-johnson , sorry for being so slow! I just added the test to python_bindings/correctness/buffer.py, is it how it supposed to look like?

**Comment by steven-johnson** [comment]
LGTM. Let me run it thru the buildbots.

**Comment by steven-johnson** [comment]
Thanks for the fix!

**Pull request merged**